test/e2e/RoleAuthorizerE2E.t.sol

```diff
@@ -0,0 +1,172 @@
+// SuT
+import {
+    BountyManager,
+    IBountyManager,
+    IERC20PaymentClient
+} from "src/modules/logicModule/BountyManager.sol";
+
+import {StreamingPaymentProcessor} from
+    "src/modules/paymentProcessor/StreamingPaymentProcessor.sol";
+
+import {
+    IStreamingPaymentProcessor,
+    IERC20PaymentClient
+} from "src/modules/paymentProcessor/IStreamingPaymentProcessor.sol";
+
+// Mocks
+import {ERC20Mock} from "test/utils/mocks/ERC20Mock.sol";
+
+contract RoleAuthorizerE2E is E2eTest {
+    address orchestratorOwner = makeAddr("orchestratorOwner");
+    address orchestratorManager = makeAddr("orchestratorManager");
+    address bountySubmitter = makeAddr("bountySubmitter");
+
+    // Constants
+    ERC20Mock token = new ERC20Mock("Mock", "MOCK");
+
+    function test_e2e_RoleAuthorizer() public {
+        // -----------INIT
+        // address(this) creates a new orchestrator.
+        IOrchestratorFactory.OrchestratorConfig memory orchestratorConfig =
+        IOrchestratorFactory.OrchestratorConfig({
+            owner: address(this),
+            token: token
+        });
+
+        IOrchestrator orchestrator =
+            _createNewOrchestratorWithAllModules(orchestratorConfig);
+
+        RebasingFundingManager fundingManager =
+            RebasingFundingManager(address(orchestrator.fundingManager()));
+
+        RoleAuthorizer authorizer =
+            RoleAuthorizer(address(orchestrator.authorizer()));
+
+        // Find BountyManager
+        BountyManager bountyManager;
+
+        address[] memory modulesList = orchestrator.listModules();
+        for (uint i; i < modulesList.length; ++i) {
+            try IBountyManager(modulesList[i]).isExistingBountyId(0) returns (
+                bool
+            ) {
+                bountyManager = BountyManager(modulesList[i]);
+                break;
+            } catch {
+                continue;
+            }
+        }
+
+        //--------------------------------------------------------------------------------
+        // Assign Bounty Manager Roles
+        //--------------------------------------------------------------------------------
+
+        // we authorize the owner to create  bounties
+        bountyManager.grantModuleRole(
+            bountyManager.BOUNTY_ISSUER_ROLE(), address(orchestratorOwner)
+        );
+
+        // we authorize the manager to verify bounty claims
+        bountyManager.grantModuleRole(
+            bountyManager.VERIFIER_ROLE(), address(orchestratorManager)
+        );
+
+        // we authorize the bountySubmitter to submit bounty claims
+        bountyManager.grantModuleRole(
+            bountyManager.CLAIMANT_ROLE(), address(bountySubmitter)
+        );
+
+        // we grant manager role to managerAddress
+        bytes32 managerRole = authorizer.getManagerRole();
+        authorizer.grantRole(managerRole, address(orchestratorManager));
+        authorizer.renounceRole(managerRole, address(this));
```

Comment:
Maybe comment why you have to remove the managerRole from address(this)

Reply:
Maybe even do the shuffle to the correct ownership earlier
I think having a clear view of who has access rights right at the start makes more sense

Reply:
added a comment. We cannot renounce ownership before assigning manager role without using prank(), that's why I it's in that order